### PR TITLE
Add micro bosh setup template

### PR DESCRIPTION
**Pull request by qingfuwang**

### CODE REVIEW

File: microbosh-setup/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    "nicName": "[parameters('vmName')]",
+    "publicIPAddressName": "[parameters('vmName')]",
+    "addressPrefix": "10.0.0.0/16",
+    "subnet1Name": "Bosh",
+    "subnet2Name": "CloudFoundry",
+    "subnet1Prefix": "10.0.0.0/24",
+    "subnet2Prefix": "10.0.16.0/24",
+    "vmStorageAccountContainerName": "vhds",
+    "publicIPAddressType": "Dynamic",
+    "networksercuritygroup_name": "bosh",
```
Comment (johngossman) [inline microbosh-setup/azuredeploy.json]: nit: misspelled. Should be networksecuritygroup_name

File: microbosh-setup/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Setup Micro Bosh Deployment VM
+
+<a href="https://azuredeploy.net/" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+
+This template allows you to setup cloud foundry development environment. It will create a vm with public ip address, storage account, virtual network, 3 reserved public ip 
+
+After finish the setup, you can download stemcell for azure from http://cloudfoundry.blob.core.windows.net/stemcell/stemcell.tgz
```
Comment (johngossman) [inline microbosh-setup/README.md]: I assume this means you need to ssh into the box?

File: microbosh-setup/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    "storageAccountType": "Standard_LRS",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnet1Ref": "[concat(variables('vnetID'),'/subnets/',variables('subnet1Name'))]",
+    "storageid": "[resourceId('Microsoft.Storage/storageAccounts',parameters('newStorageAccountName'))]",
+    "imagePublisher": "Canonical",
+    "imageOffer": "UbuntuServer",
+    "ubuntuOSVersion": "14.10",
+    "api-version": "2015-05-01-preview",
+    "branch": "qingfuwang"
```
Comment (johngossman) [inline microbosh-setup/azuredeploy.json]: This needs to be updated to Azure before we merge the PR

File: microbosh-setup/micro_cf.yml [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+      nats:
+        machines: [10.0.16.4]
+        password: c1oudc0w
+        username: nats
+    nfs_server:
+      address: 10.0.16.4
+      allow_from_entries: [10.0.16.0/20]
+    ha_proxy:
+      ssl_pem: |
+        -----BEGIN CERTIFICATE-----
```
Comment (johngossman) [inline microbosh-setup/micro_cf.yml]: I assume you don't want to check these in

File: microbosh-setup/micro_cf.yml [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+      servicesmgmt:
+        authorities: uaa.resource,oauth.service,clients.read,clients.write,clients.secret
+        authorized-grant-types: authorization_code,client_credentials,password,implicit
+        autoapprove: true
+        redirect-uri: http://servicesmgmt.cf.azurelovecf.com/auth/cloudfoundry/callback
+        scope: openid,cloud_controller.read,cloud_controller.write
+        secret: c1oudc0w
+    jwt:
+      signing_key: |
+        -----BEGIN RSA PRIVATE KEY-----
```
Comment (johngossman) [inline microbosh-setup/micro_cf.yml]: Is this meant to be checked in?

File: microbosh-setup/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+        "publisher": "Microsoft.OSTCExtensions",
+        "type": "CustomScriptForLinux",
+        "typeHandlerVersion": "1.2",
+        "settings": {
+          "fileUris": [
+            "[concat('https://raw.githubusercontent.com/', variables('branch'),'/azure-quickstart-templates/master/microbosh-setup/launch_script.sh')]"
+          ],          
+           "commandToExecute":  "[concat('sh launch_script.sh',' https://raw.githubusercontent.com/', variables('branch'),'/azure-quickstart-templates/master/microbosh-setup/')]",
+          "newStorageAccountName:":"[parameters('newStorageAccountName')]",
+          "virtualNetworkName":"[variables('virtualNetworkName')]",
```
Comment (johngossman) [inline microbosh-setup/azuredeploy.json]: I don't understand where all these settings are going.

File: microbosh-setup/metadata.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+  "itemDisplayName": "Create Cloudfoundry development machine",
```
Comment (johngossman) [inline microbosh-setup/metadata.json]: Should we say Microbosh?

File: microbosh-setup/micro_cf.yml [hunk 8]
```diff
@@ -0,0 +1,375 @@
+# The following line helps maintain current documentation at http://docs.cloudfoundry.org.
+# code_snippet cf-minimal-azure start
+---
+name: cf
+director_uuid: bfb078f7-99bc-4199-aecc-9fbb99f526b3 # Replace it with the UUID of your MicroBosh
```
Comment (johngossman) [inline microbosh-setup/micro_cf.yml]: should we leave this blank?

File: microbosh-setup/micro_cf.yml [hunk 9]
```diff
@@ -0,0 +1,375 @@
...
+releases:
+- {name: cf, version: latest}
+
+networks:
+- name: cf_private
+  type: manual
+  subnets:
+  - range: 10.0.16.0/20
+    gateway: 10.0.16.1
+    dns: [10.0.0.4]  # Replace it with your own dns ip
```
Comment (johngossman) [inline microbosh-setup/micro_cf.yml]: Should we leave this blank?

File: microbosh-setup/micro_cf.yml (hunk 4 above)
Comment (qingfuwang) [inline microbosh-setup/micro_cf.yml]: I have double checked with abel, he said this ssl is common to bosh development. I have his promise , so it's ok to check in them @AbelHu

File: microbosh-setup/micro_cf.yml (hunk 5 above)
Comment (qingfuwang) [inline microbosh-setup/micro_cf.yml]: I have double checked with abel, he said this ssl is common to bosh development. I have his promise , so it's ok to check in them @AbelHu

File: microbosh-setup/azuredeploy.json (hunk 6 above)
Comment (qingfuwang) [inline microbosh-setup/azuredeploy.json]: yes it's not the normal way to use customscript, but as our team write linux extension, so I can do something like hacking

File: microbosh-setup/micro_cf.yml (hunk 8 above)
Comment (qingfuwang) [inline microbosh-setup/micro_cf.yml]: I will left empty.

File: microbosh-setup/micro_cf.yml (hunk 9 above)
Comment (qingfuwang) [inline microbosh-setup/micro_cf.yml]: i think we can leave it as default value

File: microbosh-setup/README.md (hunk 2 above)
Comment (qingfuwang) [inline microbosh-setup/README.md]: yes

File: microbosh-setup/azuredeploy.json (hunk 6 above)
Comment (madhana) [inline microbosh-setup/azuredeploy.json]: I guess the question is why are you passing these into the extension. Does WALA use this information in the logs for better debugging?

Comment (qingfuwang) [inline microbosh-setup/azuredeploy.json]: the custom script expose one interface , commandToExecute, and my task need a lot of settings ,  it's more easy to fetch the settings in this way

**Pull request merged**